official/transformer/utils/schedule_test.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Test Transformer's schedule manager."""
+
+import tensorflow as tf
+
+from official.transformer.utils import schedule
+
+
+class ScheduleBaseTester(tf.test.TestCase):
+  def test_mutual_exclusivity(self):
+    with self.assertRaises(ValueError):
+      schedule.Manager(
+          train_steps=100,
+          steps_between_evals=100,
+          train_epochs=2,
+          epochs_between_evals=1,
+          default_train_epochs=None,
+          batch_size=2048,
+          max_length=256
+      )
+
+  def test_step_basis(self):
+    manager = schedule.Manager(
+        train_steps=1000,
+        steps_between_evals=100,
+        train_epochs=None,
+        epochs_between_evals=None,
+        default_train_epochs=None,
+        batch_size=2048,
+        max_length=256
+    )
+
```

Comment:
nit: Don't get me wrong, I love whitespace and all, but this seems like a lot of whitespace.

Reply:
Yeah, ok. Does this mean we're not going to be rewriting the codebase in Whitespace?

Reply:
I am partial to [Piet](http://www.dangermouse.net/esoteric/piet/samples.html)